### DISCUSSION TITLE
Scheduled Posts

### DISCUSSION THREAD

**Discussion by HitArrowLegend** [wintercms/wn-blog-plugin#8] [Ideas]
Hello, 

Scheduling posts is already possible in the sense that you can pick a date in the future to publish a blog post.
The thing is that a post for which the published date has been set in the future appears as published, even though it's not. 
Using the filter to only display published posts even makes this post scheduled to be published in the future appear there, while it's not technically published yet. 

So I think that the difference should be made clearer between posts that are already published, and those who are scheduled to be published, at a later date. 

What I propose is to have an additional state for posts simply called scheduled, in addition to published and non-published.
Then, a filter to only display scheduled posts so that it's easy to go through them.

**Comment by bennothommo** [comment]
@HitArrowLegend unpublished posts only appear for users logged in on the Backend with the necessary permissions. They won't appear for normal users. You could adjust your template to add a class or label to the post if its visible for a logged in backend user and has not yet been published by using this line in the template:

```
{% if date(post.published_at) < date('now') %}
...
{% endif %}
```

**Reply by LukeTowers** [reply]
@bennothommo what ever happened to the automatic inclusion of the lock emoji to the titles of not-yet-publicly viewable posts?

**Reply by HitArrowLegend** [reply]
I should have mentionned I was refering to the backend only, when im looking at the whole list of posts.

For a small list of posts it would be QoL, but for a bigger amount, this would be a much welcome change. (I'll be running a music website and with interviews, album releases, album anniversaries etc... Pretty much the first 25 posts will all be scheduled..)

**Reply by bennothommo** [reply]
@LukeTowers was that a thing? I don't think I ever saw it :/

**Reply by bennothommo** [reply]
@HitArrowLegend gotcha. Yes, I think we need to indicate that in some fashion on the backend then - perhaps using this lock icon that @LukeTowers mentioned.

**Reply by LukeTowers** [reply]
Ah, it was a thing for a client site I added:
```php
    public function afterFetch()
    {
        if (!\App::runningInBackend() && !$this->isReady()) {
            $this->title = '🔒' . $this->title;
        }
    }
```
would probably do it with an accessor though
